android/app/src/main/java/com/reactnativenavigation/controllers/NavigationActivity.java

```diff
@@ -53,6 +53,7 @@
      * Along with that, we should handle commands from the bridge using onNewIntent
      */
     static NavigationActivity currentActivity;
+    private static int activityCount;
```

Comment:
Why this needs to be static?

Reply:
So that the counter is shared between all NavigationActivity instances, and destroy only when the last one is destroyed.